de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/nodes/JsonHandler.java

```diff
@@ -67,22 +77,53 @@ public void applyJoinRelation(FskPortObject fskObj, List<JoinRelationAdvanced> j
             String resourcePath = sourceModel.getGeneratedResourcesDirectory().get()
                 .getAbsolutePath().replaceAll("\\\\", "/") + "/";
             String jsonPath = resourcePath + JSON_FILE_NAME;
-            loadParametersIntoWorkspace(jsonPath, sourceParameter, param.getId());
-
-            // if target parameter is of type FILE, add path to generatedResources to sourceParam
-            // This should be safe since source and target parameter must have the same type (if
-            // File)
-            if (param.getDataType().equals(Parameter.DataTypeEnum.FILE)) {
-              addPathToFileParameter(param.getId(), resourcePath);
+            if(!sourceTargetPathMap.containsKey(jsonPath)) {
+              sourceTargetPathMap.put(jsonPath, new ArrayList<JoinerObject>());
             }
 
-            applyJoinCommand(param.getId(), joinRelation.replaceCommand(param.getId()));
+            sourceTargetPathMap.get(jsonPath).add(new JoinerObject(sourceParameter,param,joinRelation, resourcePath));
+
           }
         }
+      }// for joinRelation
+
+
+      // work through Map, each entry is a parameters.json file, load data into workspace
+      for (Map.Entry<String, List<JoinerObject>> entry : sourceTargetPathMap.entrySet()) {
+        //ParameterData parameterData = MAPPER.readValue(new File(jsonPath), ParameterData.class);
+        ParameterData parameterData = MAPPER.readValue(new FileInputStream(entry.getKey()), ParameterData.class);

```

Comment:
FileInputStream is not closed. This could be either opened in a try-with-resources statement or just replaced with File.
I have a question about jsonPath. If they are only used here, could they be replaced with File objects? It would save you handling the file paths and you could just iterated through file keys.